lib/linter/linter.js

```diff
@@ -254,16 +256,30 @@ function createLintingProblem(options) {
         ruleId = null,
         loc = DEFAULT_ERROR_LOC,
         message = createMissingRuleMessage(options.ruleId),
-        severity = 2
+        severity = 2,
+
+        // fallback for eslintrc mode
+        language = {
+            columnStart: 0,
+            lineStart: 1
+        }
     } = options;
 
+    /*
+     * If the location returned from the parser is 0-indexed,
+     * adjust it to be 1-indexed. We always want ESLint reporting lines and
+     * columns starting from 1.
+     */
+    const columnOffset = language.columnStart === 1 ? 0 : 1;
+    const lineOffset = language.lineStart === 1 ? 0 : 1;
```

Comment:
Should we maybe use `language.lineStart` and `language.columnStart` in `report-translator` too?

https://github.com/eslint/eslint/blob/389744be255717c507fafc158746e579ac08d77e/lib/linter/report-translator.js#L250-L266

Reply:
Also in `apply-disable-directives`:

https://github.com/eslint/eslint/blob/389744be255717c507fafc158746e579ac08d77e/lib/linter/apply-disable-directives.js#L391-L392